### PR TITLE
SPEC-343: Batchsize in find test is flexible for driver

### PR DESCRIPTION
**Pull request by durran**
@behackett @jyemin @craiggwilson @rstam

### CODE REVIEW

File: source/command-monitoring/tests/README.rst [hunk 1]
```diff
@@ -50,6 +50,14 @@ The server is inconsistent on whether the ok values returned are integers or dou
 for simplicity the tests specify all expected values as doubles. Server 'ok' values of
 integers MUST be converted to doubles for comparison with the expected values.
 
+Batch Size
+^^^^^^^^^^
+
+On server versions 3.1 and higher, batch size when used in conjunction with limit is no
```
Comment (craiggwilson) [inline source/command-monitoring/tests/README.rst]: Use 3.2 everywhere except the tests themselves. Later, we won't need to come back and change text as well.

**Comment by jyemin** [comment]
The find/getMore command spec should define correct driver behavior for batch size.  I don't see why the command monitoring spec should try to work around that omission.

**Comment by craiggwilson** [comment]
Are you saying to punt on this and force the find/getMore spec to define how to deal with batchSize? If so, I agree with that statement. This isn't something that can be left unspecified. All drivers should work the same way on this.

**Comment by christkv** [comment]
that specification is limited to the 3.2 specifics. If this needs to be documented it should be in the context of a pre 3.2 wire protocol specification separate from that one.

**Comment by jyemin** [comment]
@craiggwilson, yes.

@christkv, this is about 3.2 behavior.  Through these tests we've discovered that drivers are sending different values for batchSize to the getMore command.   The Java driver is passing this test by sending a batchSize of 3 to getMore, relying on the server to enforce the limit.  The Python driver is sending a batchSize of 1 to getMore.

The end result is exactly the same: The server sends back one document and closes the cursor.

**Comment by christkv** [comment]
@jyemin I don't follow this. according to that test somehow you set a batchSize of 42 between the find and getMore command forcing the cursor to close as it returns the remaining documents. that seems like you have to manually change the batchSize on the cursor after the find command cursor was executed.

Does the java driver only allow to set batchSize at cursor execution ? 

Is your suggestion that the spec should mandate the interaction between a cursor and the batchSize ?

**Comment by jyemin** [comment]
@christkv, 42 is the test's way of telling the test runner to ignore the value.  No driver would actually send 42.

Yes, my suggestion is that the find/getMore command spec mandate what value to send for batchSize in the getMore command.  Otherwise, we have to fudge the value in this test since different drivers send different values (some send 1, some send 3, but either way the server returns 1 and closes the cursor).

**Comment by christkv** [comment]
if we have to mandate one the batchSize should be set exactly as pre 3.2 to ensure consistent behavior in drivers between 3.2 and pre 3.2. That means the value 1.

**Comment by craiggwilson** [comment]
I think that needs to be specified. You don't even have to say anything about pre-3.2 behavior. Just specify what the batchSize should be and how to calculate it (if necessary). All I'm saying is that this shouldn't be left up every driver.

**Comment by durran** [comment]
So we change this to `1` and open a spec ticket to change the find/getmore spec to specify how the batch size is calculated then? That sounds good to me - Ruby is in the same area as Python here - we are sending 1 as well as we did not change the behaviour in the calculation.

**Comment by jyemin** [comment]
I opened SPEC-351 to track my requested change to the find command spec. If that is accepted,  we still have to do something like this pull request.

File: source/command-monitoring/command-monitoring.rst [hunk 2]
```diff
@@ -381,3 +381,6 @@ Changelog
   - Changed find test with a kill cursors to not run on server versions greater than 3.0
   - Added a find test with no kill cursors command which only runs on 3.1 and higher.
   - Added notes on which tests should run based on server versions.
+
+6 OCT 2015:
+  - Changed batch size in the find test on 3.1 and higher to {{42}} with explaination in the test readme.
```
Comment (behackett) [inline source/command-monitoring/command-monitoring.rst]: explaination -> explanation

**Pull request closed without merging**